day2/gift.py

```diff
@@ -1,15 +1,22 @@
 class Gift:
     def calculation_of_wrapping_paper(self, dimensions):
-        result = 0
+        total = 0
         dimensions_new = []
         for dimension in list(dimensions):
             dimension = dimension.strip()
             dimensions_new.append(dimension)
 
         for dimension in dimensions_new:
+            result = []
             length, width, height = list(map(int, dimension.split('x')))
-            result += (
-                ((2 * length * width) +
-                 (2 * width * height) +
-                 (2 * height * length)))
-        return result
+
+            result.append(length * width)
+            result.append(width * height)
+            result.append(height * length)
+            result_min = min(result)
+
+            mult_result = lambda result: [value * 2 for value in result]
+            result = mult_result(result)
+            total += sum(result) + result_min
```

Comment:
Sei lá, antes eu até entendia um pouco do que tava acontecendo, agora não mais 😸 

Abri aqui o meu advent-of-code e vi o jeito que eu tinha resolvido essa treta, e foi assim:

https://github.com/ricardoruwer/advent-of-code/blob/master/2015/day2/box.rb

Veja o método `paper_to_order`